app/code/core/Mage/Dataflow/Model/Profile.php

```diff
@@ -86,7 +86,7 @@ protected function _beforeSave()
     {
         parent::_beforeSave();
         $actionsXML = $this->getData('actions_xml');
-        if (strlen($actionsXML) < 0 &&
+        if ((string) strlen($actionsXML) < 0 &&
```

Comment:
it doesn't seem correct, wasn't it strlen((string)$actionsXML)?

Reply:
You are right, sorry. How do you prefer to change it?

```strlen((string) $actionXML)```

or

```strlen($actionXML ?? '')```

Both work.

Reply:
I'm wondering... `strlen($actionsXML) < 0` strlen can be a negative number?? how does that happen?

but also, what is this "if" doing? it's not super clear to me, it seems it's checking for an erroneous xml

I'll think about it

Reply:
```suggestion
        if ($actionsXML !== null && strlen($actionsXML) < 0 &&
```

I think I'd do it this way, if $actionsXMLis not null and strlen is negative (which is an error) and etc etc... seems correct

Reply:
I created the PR to solve the deprecate, I didn't debug the code to understand the logic behind it. PHPStan reported something but I did not analyze it.

In the past there were discussions through forums related to the fact that the strlen() function could return negative values.

Reply:
I understand but we've to fix it in the best way and most of all be sure not to change the behavior, which is what I think my last version does

Reply:
If there are other problems then they be reported, confirmed and solved in another PR. In my opinion the deprecated functionality is solved.

Reply:
Mmm i dont think the code works the way it’s written now